src/components/Menu.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+  <popover :visible="visible" :align="alignLeft" bottom @closed="$emit('closed')">
+    <div class="menu__body">
+      <transition name="slide-left" mode="out-in">
+        <div v-if="visiblePanel == 1">
+          <div class="menu__panel">
+            <div
+              v-for="button in buttons"
+              :key="button.label"
+              class="menu__option"
+              @click="button.onclick"
+              v-html="button.html"
+            />
+            <div
+              v-if="otherButtons.length > 0"
+              class="menu__option menu__option--top-bordered"
+              @click="visiblePanel = 2"
+            >
+              Other Operations
+            </div>
+            <div class="menu__option--top-bordered">
+              <slot name="first-panel-special" />
```

Comment:
The fact that we need this makes me wonder if this abstraction is at the right level.
It seems that you had to add a whole bunch of stuff only for the ColumnMenu case.
 
What about just having a Menu component with a slot to let the inside customizable?
This way, DataTypeMenu and ActionMenu stays simple, and have another component, MultiLevelMenu, which can extend the simple Menu and handle the panel sliding.